examples/low-level-api/create_vp.rs

```diff
@@ -68,28 +73,42 @@ async fn main() -> Result<()> {
   let client: ClientMap = ClientMap::new();
 
   // Issue a Verifiable Presentation with a newly created DID Document.
-  let presentation: Presentation = create_vp().await?;
+  let (presentation, issuer_doc, holder_doc): (Presentation, IotaDocument, IotaDocument) = create_vp().await?;
 
   // Convert the Verifiable Presentation to JSON and "exchange" with a verifier
   let presentation_json: String = presentation.to_json()?;
 
-  // Create a `CredentialValidator` instance to fetch and validate all
-  // associated DID Documents from the Tangle.
-  let validator: CredentialValidator<ClientMap> = CredentialValidator::new(&client);
-
   // Validate the presentation and all the credentials included in it.
   //
   // Also verify the challenge matches.
-  let validation: PresentationValidation = validator
-    .check_presentation(
-      &presentation_json,
-      VerifierOptions::new().challenge("475a7984-1bb5-4c4c-a56f-822bccd46440".to_owned()),
-    )
-    .await?;
-  println!("validation = {:#?}", validation);
-  assert!(validation.verified);
-
-  println!("Presentation Validation > {:#?}", validation);
-
-  Ok(())
+
+  // deserialize the presentation:
+  let presentation: Presentation = Presentation::from_json(&presentation_json)?;
+  // in order to validate the presentation we need to resolve the holder's and issuer's DID documents.
+
+  //Todo: extract the issuer and holder DID's from the presentation itself
+  let resolved_holder_document: ResolvedIotaDocument = client.resolve(holder_doc.id()).await?;
+  let trusted_issuer: ResolvedIotaDocument = client.resolve(issuer_doc.id()).await?;
```

Comment:
This is exactly the pattern I created in identity.ts back in the day and honestly really regretted. First extracting the DIDs of a VC/VP to then later add them to the trusted_issuer list and then verify the VC/VP. I advise that we don't reintroduce that.

Reply:
We would have a top-level method that (perhaps optionally) resolves the issuers and holder automatically but the actual validation code needs to be abstracted from resolution to enable efficient re-use of previously resolved DID documents, such as in a DIDComm scenario where one resolved the expected holder document already. Opaque caching in the resolver which was previously discussed is still not a good solution to this in my opinion, developers need control if they want it.

Reply:
@JelleMillenaar. The `ResolvedCredential` and `ResolvedPresentation` structs would fetch the necessary DID documents upon creation which is maybe more in line with what you would prefer? It is good you mentioned this as I was considering removing them. Despite this it would be useful to hear why you regretted your decision in identity.ts ?

Reply:
Its not so much about resolving the documents, although that is also a very important topic and I agree with what Craig said, but more so about the trusted_issuer list. As discussed in the meeting, it becomes messy really quickly and there are so many situations that managing the a trusted_issuer list is probably something we need to forfeit towards the application level for now. If we want to provide any level of support for this I recommend to keep it optional and lightweight.

Reply:
We definitely can't manage a `trusted_issuer` list accounting for all possible scenarios but we need to pass them to the validator somehow to validate the credential signatures. Perhaps not calling them "trusted_issuers" and just "issuer_documents" would prevent some confusion?

See also: https://github.com/iotaledger/identity.rs/pull/599#pullrequestreview-884239877